### PR TITLE
Update to distribution/distribtion in master branch

### PR DESCRIPTION
**Pull request by rhatdan**
Improvements on https://github.com/containers/podman/issues/16205

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
@mtrmac @alexlarsson Is this what you had in mind?

File: docker/reference/normalize.go [hunk 1]
```diff
@@ -89,16 +88,21 @@ func ParseDockerRef(ref string) (Named, error) {
 // needs to be already validated before.
 func splitDockerDomain(name string) (domain, remainder string) {
 	i := strings.IndexRune(name, '/')
-	if i == -1 || (!strings.ContainsAny(name[:i], ".:") && name[:i] != "localhost") {
+	if i == -1 || (!strings.ContainsAny(name[:i], ".:") && name[:i] != "localhost" && strings.ToLower(name[:i]) == name[:i]) {
```
Comment (mtrmac) [inline docker/reference/normalize.go:91]: It’s not obvious we want this.

File: docker/reference/reference.go [hunk 2]
```diff
@@ -5,7 +5,9 @@
 //
 //	reference                       := name [ ":" tag ] [ "@" digest ]
 //	name                            := [domain '/'] path-component ['/' path-component]*
-//	domain                          := domain-component ['.' domain-component]* [':' port-number]
+//	domain                          := host [':' port-number]
+//	host                            := domain-name | IPv4address | \[ IPv6address \]	; rfc3986 appendix-A
```
Comment (mtrmac) [inline docker/reference/reference.go:9]: It’s _not at all_ obvious that we want this. c/image/storage uses a `[storageOptions]name` syntax already, how do we disambiguate?

**Review by mtrmac — COMMENTED** [review]
Maaaayyybe? It’s not at all obvious we can safely update all of it, and I personally don’t want to deal with this now.

File: docker/reference/normalize.go (hunk 1 above)
Comment (rhatdan) [reply docker/reference/normalize.go:91]: Do you think we have tests to reveal that we want this?  If not I am fine with just removing the change.

File: docker/reference/reference.go (hunk 2 above)
Comment (rhatdan) [reply docker/reference/reference.go:9]: No idea, I was just force updating, I have no problem removing the host definition from this.

Comment (mtrmac) [reply docker/reference/reference.go:9]: … and then we need to somehow manage the divergence from the upstream package, if one emerges.

I’m fine with that happening in principle, I just can’t prioritize this.

Comment (mtrmac) [reply docker/reference/reference.go:9]: A possible path forward: Update not to the head of the branch, but to some specific commit that contains exactly what we want, but nothing that we would have to think about. That way we would get the improvements we want, we would stay in sync, and we could still bring in the other changes when there’s more capacity.

_If_ such a commit exists.

Alternatively, get this reviewed by someone else who can investigate in enough detail.

Comment (mtrmac) [reply docker/reference/reference.go]: That is an effective fork, selectively choosing features… It’s an option, sure, but it turns the cost of making a decision on those other changes into technical debt to be paid later. When will that “later” ever happen?

**Comment by rhatdan** [comment]
Will go with @vrothberg version.

**Pull request closed without merging**